File: FinalProject/app/src/main/java/com/example/finalproject/fragments/MapFragment.java
package com.example.finalproject.fragments;

import android.content.Context;
import android.graphics.Bitmap;
import android.graphics.Color;
import android.net.ConnectivityManager;
import android.net.NetworkInfo;
import android.os.Bundle;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentManager;
import androidx.fragment.app.FragmentTransaction;

import android.util.Pair;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TableLayout;
import android.widget.TableRow;
import android.widget.Toast;

import com.example.finalproject.R;
import com.example.finalproject.entities.Enemy;

import java.util.Random;

public class MapFragment extends Fragment {
    private final int mapNum;

    public MapFragment(int mapNum){
        this.mapNum = mapNum;
    }

    MapFragment(){
        mapNum =0;
        MainActivity.player.setMapNum(0);
    }

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
        return inflater.inflate(R.layout.fragment_map, container, false);
    }

    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);
        Pair<Integer, Integer> player_coords=MainActivity.player.getCoordinates(mapNum);
        TableLayout table=getView().findViewById(R.id.tableLayout);
        TableRow[] rows=new TableRow[table.getChildCount()];
        for (int i=0;i<table.getChildCount();i++)
            rows[i]=(TableRow)table.getChildAt(i);
        ImageView[][] visible_map=new ImageView[5][5];
        int idLocation=512;
        View.OnClickListener onClickListener=new View.OnClickListener() {
            @Override
            public void onClick(View v) {Pair<Integer, Integer> coords= find_title_coordinates((ImageView)v, visible_map),
                    player_coords=MainActivity.player.getCoordinates(mapNum);
                if (!MainActivity.player.getCoordinates(mapNum).equals(coords)
                        &&MainActivity.map.get(mapNum).getMap()[coords.first][coords.second].getId()!=idLocation
                        &&MainActivity.map.get(mapNum).getMap()[coords.first][coords.second].getId()!=255+idLocation)
                {
                    int dx=coords.first-player_coords.first,
                            dy=coords.second-player_coords.second;
                    if (Math.abs(dx)<=1&&Math.abs(dy)<=1){
                        MainActivity.player.regenerate();
                        int a=new Random().nextInt(100);
                        int coordsId=MainActivity.map.get(mapNum).getMap()[coords.first][coords.second].getId();
                        if (coordsId!=512&& mapNum!=1 &&a<MainActivity.chancesOfFight.get(coordsId)) {
                            FragmentManager fm=getParentFragmentManager();
                            FragmentTransaction fragmentTransaction= fm.beginTransaction();
                            fragmentTransaction.remove(fm.findFragmentById(R.id.map));
                            fragmentTransaction.remove(fm.findFragmentById(R.id.menu));
                            fragmentTransaction.remove(fm.findFragmentById(R.id.status));
                            fragmentTransaction.add(R.id.fight, new FightFragment());
                            fragmentTransaction.commit();
                            a=new Random().nextInt(101);
                            switch (coordsId){
                                case 513:{
                                    if (a<30)
                                        MainActivity.player.setEnemy(
                                                new Enemy(
                                                        MainActivity.chancesOfEnemy.get(1+idLocation).get(30)));
                                    else MainActivity.player.setEnemy(
                                            new Enemy(
                                                    MainActivity.chancesOfEnemy.get(1+idLocation).get(70)));
                                    break;
                                }
                                case 514:{
                                    if (a<60)
                                        MainActivity.player.setEnemy(
                                                new Enemy(
                                                        MainActivity.chancesOfEnemy.get(2+idLocation).get(60)));
                                    else if (a<95)
                                        MainActivity.player.setEnemy(
                                                new Enemy(
                                                        MainActivity.chancesOfEnemy.get(2+idLocation).get(35)));
                                    else MainActivity.player.setEnemy(new Enemy(MainActivity.chancesOfEnemy.get(2+idLocation).get(5)));
                                    break;
                                }
                                case 516:{
                                    if (a<75)
                                        MainActivity.player.setEnemy(
                                                new Enemy(
                                                        MainActivity.chancesOfEnemy.get(4+idLocation).get(75)));
                                    else if (a<95)
                                        MainActivity.player.setEnemy(
                                                new Enemy(
                                                        MainActivity.chancesOfEnemy.get(4+idLocation).get(20)));
                                    else if (a<99)
                                        MainActivity.player.setEnemy(
                                                new Enemy(
                                                        MainActivity.chancesOfEnemy.get(4+idLocation).get(4)));
                                    else MainActivity.player.setEnemy(
                                                new Enemy(
                                                        MainActivity.chancesOfEnemy.get(4+idLocation).get(1)));
                                    break;
                                }
                                case 517:{
                                    if (a<75)
                                        MainActivity.player.setEnemy(
                                                new Enemy(
                                                        MainActivity.chancesOfEnemy.get(5+idLocation).get(75)));
                                    else if (a<95)
                                        MainActivity.player.setEnemy(
                                                new Enemy(
                                                        MainActivity.chancesOfEnemy.get(5+idLocation).get(20)));
                                    else if (a<99)
                                        MainActivity.player.setEnemy(
                                                new Enemy(
                                                        MainActivity.chancesOfEnemy.get(5+idLocation).get(4)));
                                    else MainActivity.player.setEnemy(
                                                new Enemy(
                                                        MainActivity.chancesOfEnemy.get(5+idLocation).get(1)));
                                    break;
                                }
                                case 518:{
                                    if (a<60)
                                        MainActivity.player.setEnemy(
                                                new Enemy(
                                                        MainActivity.chancesOfEnemy.get(6+idLocation).get(60)));
                                    else MainActivity.player.setEnemy(
                                            new Enemy(
                                                    MainActivity.chancesOfEnemy.get(6+idLocation).get(40)));
                                    break;
                                }
                                case 519:{
                                    if (a<60)
                                        MainActivity.player.setEnemy(
                                                new Enemy(
                                                        MainActivity.chancesOfEnemy.get(7+idLocation).get(60)));
                                    else MainActivity.player.setEnemy(
                                            new Enemy(
                                                    MainActivity.chancesOfEnemy.get(7+idLocation).get(40)));
                                    break;
                                }
                            }
                        } else{
                            switch (coordsId){
                                case 3+512:{
                                    MainActivity.player.setMapNum(1);
                                    MainActivity.player.setCoordinates(1, new Pair<>(6, 3));
                                    FragmentManager fm = getParentFragmentManager();
                                    FragmentTransaction fragmentTransaction = fm.beginTransaction();
                                    fragmentTransaction.remove(fm.findFragmentById(R.id.map));
                                    fragmentTransaction.add(R.id.map, new MapFragment(1));
                                    fragmentTransaction.commit();
                                    break;
                                }
                                case 9+512:{
                                    FragmentManager fm = getParentFragmentManager();
                                    FragmentTransaction fragmentTransaction = fm.beginTransaction();
                                    fragmentTransaction.remove(fm.findFragmentById(R.id.map));
                                    fragmentTransaction.remove(fm.findFragmentById(R.id.status));
                                    fragmentTransaction.remove(fm.findFragmentById(R.id.menu));
                                    fragmentTransaction.add(R.id.tasks, new TaskManagerFragment(true));
                                    fragmentTransaction.commit();
                                    break;
                                }
                                case 10+512:{
                                    if (isInternetAvailable()){
                                        if (MainActivity.player.getUser().getLogin().isEmpty())
                                            Toast.makeText(getContext(), "Sign in first", Toast.LENGTH_SHORT).show();
                                        else {
                                            FragmentManager fm = getParentFragmentManager();
                                            FragmentTransaction fragmentTransaction = fm.beginTransaction();
                                            fragmentTransaction.remove(fm.findFragmentById(R.id.map));
                                            fragmentTransaction.remove(fm.findFragmentById(R.id.status));
                                            fragmentTransaction.remove(fm.findFragmentById(R.id.menu));
                                            fragmentTransaction.add(R.id.fight, new FightFragment(true));
                                            fragmentTransaction.commit();
                                        }
                                    }
                                    else Toast.makeText(getContext(), "Check your Internet connection", Toast.LENGTH_SHORT).show();
                                    break;
                                }
                                case 11+512:{
                                    FragmentManager fm = getParentFragmentManager();
                                    FragmentTransaction fragmentTransaction = fm.beginTransaction();
                                    fragmentTransaction.remove(fm.findFragmentById(R.id.map));
                                    fragmentTransaction.remove(fm.findFragmentById(R.id.status));
                                    fragmentTransaction.remove(fm.findFragmentById(R.id.menu));
                                    fragmentTransaction.add(R.id.crafting_station, new CraftingStationFragment());
                                    fragmentTransaction.commit();
                                    break;
                                }
                                case 12+512:{
                                    FragmentManager fm = getParentFragmentManager();
                                    FragmentTransaction fragmentTransaction = fm.beginTransaction();
                                    fragmentTransaction.remove(fm.findFragmentById(R.id.map));
                                    fragmentTransaction.remove(fm.findFragmentById(R.id.status));
                                    fragmentTransaction.remove(fm.findFragmentById(R.id.menu));
                                    fragmentTransaction.add(R.id.shop, new ShopFragment());
                                    fragmentTransaction.commit();
                                    break;
                                }
                                case 13+512:{
                                    FragmentManager fm = getParentFragmentManager();
                                    FragmentTransaction fragmentTransaction = fm.beginTransaction();
                                    fragmentTransaction.remove(fm.findFragmentById(R.id.map));
                                    fragmentTransaction.add(R.id.map, new MapFragment());
                                    fragmentTransaction.commit();
                                    break;
                                }
                                case 14+512:{
                                    if (isInternetAvailable()){
                                        if (MainActivity.player.getUser().getLogin().isEmpty()){
                                            Toast.makeText(getContext(), "Sign in first", Toast.LENGTH_SHORT).show();
                                        }
                                        else {
                                            FragmentManager fm = getParentFragmentManager();
                                            FragmentTransaction fragmentTransaction = fm.beginTransaction();
                                            fragmentTransaction.remove(fm.findFragmentById(R.id.map));
                                            fragmentTransaction.remove(fm.findFragmentById(R.id.status));
                                            fragmentTransaction.remove(fm.findFragmentById(R.id.menu));
                                            fragmentTransaction.add(R.id.shop, new ShopFragment(true));
                                            fragmentTransaction.commit();
                                        }
                                    }
                                    else Toast.makeText(getContext(), "Check your Internet connection", Toast.LENGTH_SHORT).show();
                                    break;
                                }
                            }
                        }
                        if (coordsId>526||coordsId<520)
                        {
                            MainActivity.player.setTileTexture(
                                    MainActivity.mapTextures.get(
                                            MainActivity.map.get(mapNum)
                                                    .getMap()[player_coords.first][player_coords.second].getId()-idLocation));
                            MainActivity.map.get(mapNum).getMap()[player_coords.first][player_coords.second].setTexture(
                                    Bitmap.createBitmap(MainActivity.player.getTileTexture()));
                            MainActivity.player.setCoordinates(mapNum,
                                    new Pair<>(player_coords.first+dx, player_coords.second+dy));
                            player_coords=MainActivity.player.getCoordinates(mapNum);
                            MainActivity.player.setTileTexture(
                                    Bitmap.createBitmap(
                                            MainActivity.map.get(mapNum)
                                                    .getMap()[player_coords.first][player_coords.second].getTexture()));
                            MainActivity.map.get(mapNum).getMap()[player_coords.first][player_coords.second]
                                    .getTexture().eraseColor(Color.BLUE);
                            if (player_coords.first+dx>=2&&player_coords.second+dy>=2){
                                for (int i=0;i<5;i++) {
                                    for (int j = 0; j < 5; j++) {
                                        visible_map[i][j].setImageBitmap(
                                                MainActivity.map.get(mapNum).getMap()[player_coords.first-2+i]
                                                        [player_coords.second-2+j].getTexture());
                                    }
                                }
                            }
                            else if (player_coords.first+dx>=2) {
                                for (int i=0;i<5;i++) {
                                    for (int j = 0; j < 5; j++) {
                                        visible_map[i][j].setImageBitmap(
                                                MainActivity.map.get(mapNum).getMap()[player_coords.first-2+i][j].getTexture());
                                    }
                                }
                            }
                            else if (player_coords.second+dy>=2){
                                for (int i=0;i<5;i++) {
                                    for (int j = 0; j < 5; j++) {
                                        visible_map[i][j].setImageBitmap(
                                                MainActivity.map.get(mapNum).getMap()[i][player_coords.second-2+j].getTexture());
                                    }
                                }
                            }
                            else {
                                for (int i=0;i<5;i++) {
                                    for (int j = 0; j < 5; j++) {
                                        visible_map[i][j].setImageBitmap(MainActivity.map.get(mapNum).getMap()[i][j].getTexture());
                                    }
                                }
                            }
                        }
                    }
                }
                StatusBarFragment.update();
            }
        };
        for (int i=0;i<5;i++) {
            for (int j = 0; j < 5; j++) {
                visible_map[i][j] = (ImageView) rows[i].getChildAt(j);
                visible_map[i][j].setImageBitmap(MainActivity.map.get(mapNum)
                        .getMap()[player_coords.first-2+i]
                        [player_coords.second-2+j].getTexture());
                visible_map[i][j].setOnClickListener(onClickListener);
            }
        }
        MainActivity.player.setTileTexture(
                Bitmap.createBitmap(MainActivity.map.get(mapNum).
                        getMap()[MainActivity.player.getCoordinates(mapNum).first]
                        [MainActivity.player.getCoordinates(mapNum).second].getTexture()));
        MainActivity.map.get(mapNum).getMap()[MainActivity.player.getCoordinates(mapNum).first]
                [MainActivity.player.getCoordinates(mapNum).second]
                .getTexture().eraseColor(Color.BLUE);
    }

    private Pair<Integer, Integer> find_title_coordinates(ImageView v, ImageView[][] p){
        for (int i=0;i<5;i++){
            for (int j=0;j<5;j++){
                if (v==p[i][j])
                    return new Pair<>(i+MainActivity.player.getCoordinates(mapNum).first-2,
                            j+MainActivity.player.getCoordinates(mapNum).second-2);
            }
        }
        return null;
    }

    private boolean isInternetAvailable() {
        ConnectivityManager cm =
                (ConnectivityManager)getContext().getSystemService(Context.CONNECTIVITY_SERVICE);
        NetworkInfo activeNetwork = cm.getActiveNetworkInfo();
        return activeNetwork != null &&
                activeNetwork.isConnectedOrConnecting();
    }
}

File: FinalProject/app/src/main/java/com/example/finalproject/fragments/MapFragment.kt
package com.example.finalproject.fragments

import android.content.Context
import android.graphics.Bitmap
import android.net.ConnectivityManager
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.TableLayout
import android.widget.TableRow
import android.widget.Toast
import androidx.fragment.app.Fragment
import com.example.finalproject.R
import com.example.finalproject.entities.Enemy
import kotlin.math.abs
import com.example.finalproject.fragments.MainActivity.Companion.player
import java.util.*
import kotlin.random.Random

class MapFragment(val mapNum: Int = 0) : Fragment(), View.OnClickListener {

    private val visibleMap: Array<Array<ImageView?>> = Array(5) { arrayOfNulls(5) }
    private val map = MainActivity.map[mapNum].map
    private lateinit var enemy:Enemy

    constructor() : this(0) {
        player.mapNum = 0
    }

    override fun onCreateView(
        inflater: LayoutInflater, container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View? {
        return inflater.inflate(R.layout.fragment_map, container, false)
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        val table: TableLayout = requireView().findViewById(R.id.tableLayout)
        val rows: Array<TableRow?> = arrayOfNulls(table.childCount)
        for (i in 0 until table.childCount) rows[i] = table.getChildAt(i) as TableRow
        for (i in 0..4) {
            for (j in 0..4) {
                visibleMap[i][j] = rows[i]!!.getChildAt(j) as ImageView
                visibleMap[i][j]!!.setOnClickListener(this)
                if (i == 2 && j == 2)
                    visibleMap[i][j]!!.setImageBitmap(MainActivity.getAvatar())
                else
                    visibleMap[i][j]!!.setImageBitmap(
                        Bitmap.createBitmap(
                            map[player.coordinates[mapNum].first - 2 + i]
                                    [player.coordinates[mapNum].second - 2 + j].getTexture()
                        )
                    )

            }
        }
    }

    private fun findTitleCoordinates(v: ImageView, p: Array<Array<ImageView?>>): Pair<Int, Int> {
        for (i in 0..4) {
            for (j in 0..4) {
                if (v === p[i][j])
                    return Pair(
                        i + player.coordinates[mapNum].first - 2,
                        j + player.coordinates[mapNum].second - 2
                    )
            }
        }
        return Pair(-1, -1)
    }

    private fun isInternetAvailable(): Boolean {
        val activeNetwork =
            (requireContext().getSystemService(Context.CONNECTIVITY_SERVICE)
                    as ConnectivityManager).activeNetworkInfo
        return activeNetwork?.isConnectedOrConnecting == true
    }

    override fun onClick(v: View?) {
        if (v is ImageView) {
            val idLocation = 512
            val clickCoordinates = findTitleCoordinates(v, visibleMap)
            var playerCoordinates: Pair<Int, Int> = player.coordinates[mapNum]
            if (player.coordinates[mapNum] != clickCoordinates
                && map[clickCoordinates.first][clickCoordinates.second].id != idLocation
                && map[clickCoordinates.first][clickCoordinates.second].id != 255 + idLocation
            ) {
                val dx = clickCoordinates.first - playerCoordinates.first
                val dy = clickCoordinates.second - playerCoordinates.second
                if (abs(dx) <= 1 && abs(dy) <= 1) {
                    player.regenerate()
                    var chance = Random(Date().time).nextInt(101)
                    val tileId = map[clickCoordinates.first][clickCoordinates.second].id
                    if (tileId != 512 && mapNum != 1 && chance < MainActivity.chancesOfFight[tileId]!!) {
                        chance = Random(Date().time).nextInt(101)
                        when (tileId) {
                            1 + idLocation -> {
                                enemy = if (chance < 30) Enemy(
                                    MainActivity.chancesOfEnemy[1 + idLocation]!![30]!!
                                )
                                else Enemy(
                                    MainActivity.chancesOfEnemy[1 + idLocation]!![70]!!
                                )
                            }
                            2 + idLocation -> {
                                enemy = when {
                                    chance < 60 -> Enemy(
                                        MainActivity.chancesOfEnemy[2 + idLocation]!![60]!!
                                    )
                                    chance < 95 -> Enemy(
                                        MainActivity.chancesOfEnemy[2 + idLocation]!![35]!!
                                    )
                                    else -> Enemy(
                                        MainActivity.chancesOfEnemy[2 + idLocation]!![5]!!
                                    )
                                }
                            }
                            4 + idLocation -> {
                                enemy = when {
                                    chance < 75 -> Enemy(
                                        MainActivity.chancesOfEnemy[4 + idLocation]!![75]!!
                                    )
                                    chance < 95 -> Enemy(
                                        MainActivity.chancesOfEnemy[4 + idLocation]!![20]!!
                                    )
                                    chance < 99 -> Enemy(
                                        MainActivity.chancesOfEnemy[4 + idLocation]!![4]!!
                                    )
                                    else -> Enemy(
                                        MainActivity.chancesOfEnemy[4 + idLocation]!![1]!!
                                    )
                                }
                            }
                            5 + idLocation -> {
                                enemy = when {
                                    chance < 75 -> Enemy(
                                        MainActivity.chancesOfEnemy[5 + idLocation]!![75]!!
                                    )
                                    chance < 95 -> Enemy(
                                        MainActivity.chancesOfEnemy[5 + idLocation]!![20]!!
                                    )
                                    chance < 99 -> Enemy(
                                        MainActivity.chancesOfEnemy[5 + idLocation]!![4]!!
                                    )
                                    else -> Enemy(
                                        MainActivity.chancesOfEnemy[5 + idLocation]!![1]!!
                                    )
                                }
                            }
                            6 + idLocation -> {
                                enemy = if (chance < 60) Enemy(
                                    MainActivity.chancesOfEnemy[6 + idLocation]!![60]!!
                                )
                                else Enemy(
                                    MainActivity.chancesOfEnemy[6 + idLocation]!![40]!!
                                )
                            }
                            7 + idLocation -> {
                                enemy = if (chance < 60) Enemy(
                                    MainActivity.chancesOfEnemy[7 + idLocation]!![60]!!
                                )
                                else Enemy(
                                    MainActivity.chancesOfEnemy[7 + idLocation]!![40]!!
                                )
                            }
                        }
                        val fragmentManager = parentFragmentManager
                        val fragmentTransaction = fragmentManager.beginTransaction()
                        fragmentTransaction.remove(fragmentManager.findFragmentById(R.id.map)!!)
                        fragmentTransaction.remove(fragmentManager.findFragmentById(R.id.menu)!!)
                        fragmentTransaction.remove(fragmentManager.findFragmentById(R.id.status)!!)
                        fragmentTransaction.add(R.id.fight, FightFragment(false, enemy))
                        fragmentTransaction.commit()
                    } else {
                        when (tileId) {
                            3 + idLocation -> {
                                player.mapNum = 1
                                player.coordinates[1] = Pair(6, 3)
                                val fm = parentFragmentManager
                                val fragmentTransaction = fm.beginTransaction()
                                fragmentTransaction.remove(fm.findFragmentById(R.id.map)!!)
                                fragmentTransaction.add(R.id.map, MapFragment(1))
                                fragmentTransaction.commit()
                            }
                            9 + idLocation -> {
                                val fm = parentFragmentManager
                                val fragmentTransaction = fm.beginTransaction()
                                fragmentTransaction.remove(fm.findFragmentById(R.id.map)!!)
                                fragmentTransaction.remove(fm.findFragmentById(R.id.status)!!)
                                fragmentTransaction.remove(fm.findFragmentById(R.id.menu)!!)
                                fragmentTransaction.add(R.id.tasks, TaskManagerFragment(true))
                                fragmentTransaction.commit()
                            }
                            10 + idLocation -> {
                                if (isInternetAvailable()) {
                                    if (player.user.login.isEmpty())
                                        Toast.makeText(context, "Sign in first", Toast.LENGTH_SHORT)
                                            .show()
                                    else {
                                        val fm = parentFragmentManager
                                        val fragmentTransaction = fm.beginTransaction()
                                        fragmentTransaction.remove(fm.findFragmentById(R.id.map)!!)
                                        fragmentTransaction.remove(fm.findFragmentById(R.id.status)!!)
                                        fragmentTransaction.remove(fm.findFragmentById(R.id.menu)!!)
                                        fragmentTransaction.add(R.id.fight, FightFragment(true, enemy))
                                        fragmentTransaction.commit()
                                    }
                                } else
                                    Toast.makeText(
                                        context, "Check your Internet connection",
                                        Toast.LENGTH_SHORT
                                    ).show()
                            }
                            11 + idLocation -> {
                                val fm = parentFragmentManager
                                val fragmentTransaction = fm.beginTransaction()
                                fragmentTransaction.remove(fm.findFragmentById(R.id.map)!!)
                                fragmentTransaction.remove(fm.findFragmentById(R.id.status)!!)
                                fragmentTransaction.remove(fm.findFragmentById(R.id.menu)!!)
                                fragmentTransaction.add(
                                    R.id.crafting_station,
                                    CraftingStationFragment()
                                )
                                fragmentTransaction.commit()
                            }
                            12 + idLocation -> {
                                val fm = parentFragmentManager
                                val fragmentTransaction = fm.beginTransaction()
                                fragmentTransaction.remove(fm.findFragmentById(R.id.map)!!)
                                fragmentTransaction.remove(fm.findFragmentById(R.id.status)!!)
                                fragmentTransaction.remove(fm.findFragmentById(R.id.menu)!!)
                                fragmentTransaction.add(R.id.shop, ShopFragment())
                                fragmentTransaction.commit()
                            }
                            13 + idLocation -> {
                                val fm = parentFragmentManager
                                val fragmentTransaction = fm.beginTransaction()
                                fragmentTransaction.remove(fm.findFragmentById(R.id.map)!!)
                                fragmentTransaction.add(R.id.map, MapFragment())
                                fragmentTransaction.commit()
                            }
                            14 + idLocation -> {
                                if (isInternetAvailable()) {
                                    if (player.user.login.isEmpty()) {
                                        Toast.makeText(
                                            context, "Sign in first",
                                            Toast.LENGTH_SHORT
                                        ).show()
                                    } else {
                                        val fm = parentFragmentManager
                                        val fragmentTransaction = fm.beginTransaction()
                                        fragmentTransaction.remove(fm.findFragmentById(R.id.map)!!)
                                        fragmentTransaction.remove(fm.findFragmentById(R.id.status)!!)
                                        fragmentTransaction.remove(fm.findFragmentById(R.id.menu)!!)
                                        fragmentTransaction.add(R.id.shop, ShopFragment(true))
                                        fragmentTransaction.commit()
                                    }
                                } else
                                    Toast.makeText(
                                        context, "Check your Internet connection",
                                        Toast.LENGTH_SHORT
                                    ).show()
                            }
                        }
                    }
                    if (tileId !in idLocation + 8..idLocation + 14) {
                        player.coordinates[mapNum] =
                            Pair(playerCoordinates.first + dx, playerCoordinates.second + dy)
                        playerCoordinates = player.coordinates[mapNum]
                        for (i in 0..4)
                            for (j in 0..4) {
                                visibleMap[i][j]!!.setOnClickListener(this)
                                if (i == 2 && j == 2)
                                    visibleMap[i][j]!!.setImageBitmap(MainActivity.getAvatar())
                                else
                                    visibleMap[i][j]!!.setImageBitmap(
                                        Bitmap.createBitmap(
                                            map[playerCoordinates.first - 2 + i][playerCoordinates.second - 2 + j].getTexture()
                                        )
                                    )
                            }
                    }
                }
            }
            (parentFragmentManager.findFragmentById(R.id.status) as StatusBarFragment).update()
        }
    }
}